admin/includes/modules/update_product.php

```diff
@@ -12,7 +12,7 @@
   if (isset($_GET['pID'])) $products_id = zen_db_prepare_input($_GET['pID']);
   if (isset($_POST['edit_x']) || isset($_POST['edit_y'])) {
     $action = 'new_product';
-  } elseif ($_POST['products_model'] . $_POST['products_url'] . $_POST['products_name'] . $_POST['products_description'] != '') {
+  } elseif ((isset($_POST['products_model']) ? $_POST['products_model'] : '') . (isset($_POST['products_url']) ? implode('', $_POST['products_url']) : '') . (isset($_POST['products_name']) ? implode('', $_POST['products_name']) : '') . (isset($_POST['products_description']) ? implode('', $_POST['products_description']) : '') != '') {

```

Comment:
Using `implode()` here is an interesting approach. Mind clarifying your "why"?

Reply:
Ultimately in a multi-language environment, the fields `products_url`, `products_name`, and `products_description` all are passed as an `array` to include the language identifier to finish off the posted information.   (Even in a single language situation, the source code will still present the language identifier for each of those fields.) 
So, instead of evaluating the language(s) implemented (doesn't add value to further evaluate knowing that there is always a language identifier), determining whether the value has been passed as an array (may be an issue in future PHP versions) and knowing that the `implode()` function will generate a text item representing the single text item or the list of items separated by the `glue` (an empty quote), combining that data to evaluate the presence of a character fits soundly. Perhaps some other comparison (regex?) of that combination would be a better fit but if someone wants to store a blank or space, why should that be prevented? 

But, the idea was to use the existing approach in a stricter php manner based on the type(s) of input expected.

Reply:
Thanks. I like it!